internal/extproc/watcher_test.go

```diff
@@ -40,9 +42,30 @@ func (m *mockReceiver) getConfig() *filterapi.Config {
 	return m.cfg
 }
 
+var _ io.Writer = (*syncBuffer)(nil)
+
+// syncBuffer is a bytes.Buffer that is safe for concurrent read/write access.
+// used just in the tests to safely read the logs in assertions without data races.
+type syncBuffer struct {
+	mu sync.RWMutex
+	b  *bytes.Buffer
+}
```

Comment:
I'm not sure why this didn't fail before, but when running the unit tests I got data races in the buffer, between the message logs and the call to `buf.String()`. This should fix that .